### DISCUSSION TITLE
why does select().correlate() not seem to have an effect when printed by itself?

### DISCUSSION THREAD

**Discussion by kaitelam** [sqlalchemy/sqlalchemy#8284] [Usage Questions] [answered]
### Describe the bug

Query correlation does not work anymore on the latest versions of sqlalchemy.

The code bellow is expected to produce the following sql:
```sql
SELECT child.id, parent.value 
FROM child 
WHERE child.parent_id = parent.id
```
but unfortunately, it is giving 
```sql
SELECT child.id, parent.value 
FROM child, parent 
WHERE child.parent_id = parent.id
```

### To Reproduce

```python
import sqlalchemy as sa
metadata_obj = sa.MetaData()
parent = sa.Table(
    "parent",
    metadata_obj,
    sa.Column(
        "id",
        sa.String,
    ),
    sa.Column("value", sa.String),
)
child = sa.Table(
    "child",
    metadata_obj,
    sa.Column(
        "id",
        sa.String,
    ),
    sa.Column(
        "parent_id",
        sa.String,
    )
)

s = sa.select(child.c.id, parent.c.value).where(child.c.parent_id==parent.c.id).correlate(parent)
print(s)
```

### Error

```
# Copy complete stack trace and error message here, including SQL log output if applicable.
``` 

### Versions

- OS: ubuntu 20.04
- Python: 3.8.12
- SQLAlchemy: 1.4.39
- Database:
- DBAPI (eg: psycopg, cx_oracle, mysqlclient):

**Comment by zzzeek** [comment, marked as answer]
hi there -

as mentioned in the [bug report template](https://github.com/sqlalchemy/sqlalchemy/blob/main/.github/ISSUE_TEMPLATE/bug_report.yaml), please open discussions and not issues.

A [correlated subquery](https://docs.sqlalchemy.org/en/14/glossary.html#term-correlates) is typically a single-column, scalar subquery, unless you were using the subquery wtihin a LATERAL expression.  So here, in order for this subquery to be usable under normal circumstances, it would only SELECT from a single column and not two.

Secondly, SQLAlchemy correlation only has any effect **within the context of an enclosing query**.  So any SELECT object you print will **always** render all FROM clauses, unless there is an enclosing subquery for which one or more FROM clauses can correlate.     This behavior has been this way since ... (was curious, did a check), SQLAlchemy 0.8 which was released in 2012, so for the last ten years.

to see correlation have an effect in your example:

```py
statement_will_correlate = sa.select(child.c.id, parent.c.value).where(child.c.parent_id==parent.c.id).correlate(parent)

illustration_of_correlation = sa.select(
    parent
).where(parent.c.value == statement_will_correlate.with_only_columns(parent.c.value).scalar_subquery())
print(illustration_of_correlation)
```

```sql
SELECT parent.id, parent.value 
FROM parent 
WHERE parent.value = (SELECT parent.value 
FROM child 
WHERE child.parent_id = parent.id)

```
